fish-rust/src/signal.rs

```diff
@@ -65,3 +67,272 @@ pub fn signal_get_desc(sig: i32) -> &'static wstr {
 
     wstr::from_ucstr(s).expect("signal description should be valid utf-32")
 }
+
+#[derive(Clone, Copy, Eq, PartialEq, PartialOrd, Ord)]
+/// A wrapper around the system signal code.
+pub struct Signal(NonZeroU8);
```

Comment:
Signals have type `int`, so I expect this to wrap an int.

Reply:
I originally wrapped Signal around NonZeroI32 but went with code purity instead since signal numbers just don't go that high even though they're returned as int by the system (largely because that's the smallest type the system really uses, especially since char isn't defined as being signed or unsigned). But I can change everything back to i32/NonZeroI32 to get rid of any downsizing safety checks.